halo/lua/lua_open.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+** Modified from LuaJIT to fit the sandboxed nature of the project.
+** Copyright (C) 2019 gbMichelle
+**
+** Library initialization.
+** Copyright (C) 2005-2017 Mike Pall. See Copyright Notice in luajit.h
+**
+** Major parts taken verbatim from the Lua interpreter.
+** Copyright (C) 1994-2012 Lua.org, PUC-Rio. See Copyright Notice at the bottom.
+*/
+
+#define lib_init_c
+#define LUA_LIB
+
+#include <cctype>
+#include <string>
+
+#include "../functions/messaging.hpp"
+#include "../memory/global.hpp"
+#include "../paths.hpp"
+
+#include "lua_helpers.hpp"
+#include "lua_open.hpp"
+
+// This code has been taken from the LuaJIT source code and modified to fit
+// Vulpes needs.
+
+// Sentinel against recursive loops. Taken from LuaJIT lib_package source.
+static int sentinel_ = 0;
+#define sentinel reinterpret_cast<void *>(&sentinel_)
+
+static int LuaV_sandboxed_require(lua_State *L){
+    // Get the name from the stack:
+    auto name = luaL_checkstring(L, 1);
+    // LuaJIT source (Added some comments and made an error more clear):
+    // Get the table of loaded modules:
+    lua_settop(L, 1);
+    lua_getfield(L, LUA_REGISTRYINDEX, "_LOADED");
+    // Check if the requested name is in the table:
+    lua_getfield(L, 2, name);
+    if (lua_toboolean(L, -1)) {
+        if (lua_touserdata(L, -1) == sentinel){
+            luaL_error(L, "Recursive loop or previous error loading module %s", name);
+        }
+        return 1;
+    }
+    // End of LuaJIT source^
+
+    // In sandboxed mode we're not allowing anything but alphanumerics
+    // underscores, dashes and dots for file extensions.
+    for (char & c : std::string(name)) {
+        if ((!isalnum(c)) && (c != *".") && (c != *"_") && (c != *"-")){
```

Comment:
Have you tried single quoting the characters here? Your issue might be that you're comparing `char` to `char *` here.